multicluster/controllers/multicluster/member/serviceexport_controller.go

```diff
@@ -231,19 +238,26 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			Namespace: req.Namespace,
 		},
 	}
-	readyAddresses := 0
-	err = r.Client.Get(ctx, req.NamespacedName, eps)
-	if err == nil {
-		for _, subsets := range eps.Subsets {
-			readyAddresses = readyAddresses + len(subsets.Addresses)
+
+	var hasReadyEndpoints bool
+	var newSubsets []corev1.EndpointSubset
+	if r.endpointSliceEnabled {
+		newSubsets, hasReadyEndpoints, err = r.getSubsetsFromEndpointSlice(ctx, req)
+		if err != nil {
+			return ctrl.Result{}, err
 		}
 	} else {
-		if !apierrors.IsNotFound(err) {
+		hasReadyEndpoints, err = r.checkSubsetsFromEndpoint(ctx, req, eps)
```

Comment:
why don't return newSubsets from `checkSubsetsFromEndpoint` and remove `filterEndpointSubsets`? They seems to do same work.